core/src/main/scala/kafka/server/ClientQuotaManager.scala

```diff
@@ -54,11 +59,84 @@ object ClientQuotaManagerConfig {
   val DefaultQuotaWindowSizeSeconds = 1
   // Purge sensors after 1 hour of inactivity
   val InactiveSensorExpirationTimeSeconds  = 3600
+
+  val UnlimitedQuota = Quota.upperBound(Long.MaxValue)
+  val DefaultClientIdQuotaId = QuotaId.quotaId(None, Some(ConfigEntityName.Default))
+  val DefaultUserQuotaId = QuotaId.quotaId(Some(ConfigEntityName.Default), None)
+  val DefaultUserClientIdQuotaId = QuotaId.quotaId(Some(ConfigEntityName.Default), Some(ConfigEntityName.Default))
+}
+
+object QuotaTypes {
+  val NoQuotas = 0
+  val ClientIdQuotaEnabled = 1
+  val UserQuotaEnabled = 2
+  val UserClientIdQuotaEnabled = 4
+}
+
+object QuotaId {
+
+  val Separator = ':'
```

Comment:
The separator is used only in sensor name. Metrics store user and clientId as separate tags. To avoid confusion, I have moved the concatenation to sensor names alone.

I think % is valid in JMX, but just in case some tools don't like it, I have changed sanitize method to encode underscore and then replace % with underscores. So now an encoded user principal contains the same set of characters as clientIds, which should make it valid everywhere.
`<default>` is used in Zookeeper nodes for default configuration, but never in metrics (metrics use actual clientIds and users only). Zookeeper docs (https://zookeeper.apache.org/doc/r3.1.2/zookeeperProgrammers.html#ch_zkDataModel) say that this should be valid. I have tested it too and it seems fine.